lib/Infer/EnumerativeSynthesis.cpp

```diff
@@ -114,12 +124,25 @@ namespace {
 // experiment with synthesizing at reduced bitwidth, then expanding the result
 // aggressively avoid calling into the solver
 
-void addGuess(Inst *RHS, int MaxCost, std::vector<Inst *> &Guesses,
+void addGuess(Inst *RHS, unsigned TargetWidth, InstContext &IC, int MaxCost, std::vector<Inst *> &Guesses,
               int &TooExpensive) {
-  if (souper::cost(RHS) < MaxCost)
-    Guesses.push_back(RHS);
-  else
-    TooExpensive++;
+  if (TargetWidth > RHS->Width) {
+    auto NSExt = IC.getInst(Inst::SExt, TargetWidth, { RHS });
+    auto NZExt = IC.getInst(Inst::ZExt, TargetWidth, { RHS });
+    addGuess(NSExt, TargetWidth, IC, MaxCost, Guesses, TooExpensive);
+    addGuess(NZExt, TargetWidth, IC, MaxCost, Guesses, TooExpensive);
+  }
+  else if (TargetWidth < RHS->Width) {
+    auto NTrunc = IC.getInst(Inst::Trunc, TargetWidth, { RHS });
+    addGuess(NTrunc, TargetWidth, IC, MaxCost, Guesses, TooExpensive);
+  }
+  else {
```

Comment:
plz always put else on the same line as the closing curly brace of the previous block